diff -ur sane-backends-1.0.25.orig/configure.in sane-backends-1.0.25/configure.in
--- sane-backends-1.0.25.orig/configure.in	2015-10-03 04:00:34.000000000 +0200
+++ sane-backends-1.0.25/configure.in	2017-03-19 20:13:46.000000000 +0100
@@ -169,8 +169,8 @@
 		snmp_version=`$SNMP_CONFIG_PATH --version 2>/dev/null`
 		vers=`echo $snmp_version | awk 'BEGIN { FS = "."; } { printf "%d", ($1 * 1000 + $2);}'`
 		if test -n "$vers" && test "$vers" -ge 5006; then
-			SNMP_LIBS=`$SNMP_CONFIG_PATH --libs`
-			SNMP_CFLAGS=`$SNMP_CONFIG_PATH --cflags`
+			SNMP_LIBS="-lnetsnmp"
+			SNMP_CFLAGS=
 			AC_MSG_RESULT(yes)
 			with_snmp="yes"
 		else
diff -ur sane-backends-1.0.25.orig/configure sane-backends-1.0.25/configure
--- sane-backends-1.0.25.orig/configure	2015-10-04 15:18:39.000000000 +0200
+++ sane-backends-1.0.25/configure	2017-03-19 20:16:11.000000000 +0100
@@ -8012,8 +8012,8 @@
 		snmp_version=`$SNMP_CONFIG_PATH --version 2>/dev/null`
 		vers=`echo $snmp_version | awk 'BEGIN { FS = "."; } { printf "%d", ($1 * 1000 + $2);}'`
 		if test -n "$vers" && test "$vers" -ge 5006; then
-			SNMP_LIBS=`$SNMP_CONFIG_PATH --libs`
-			SNMP_CFLAGS=`$SNMP_CONFIG_PATH --cflags`
+			SNMP_LIBS="-lnetsnmp"
+			SNMP_CFLAGS=
 			{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
 			with_snmp="yes"
