diff -ur sane-backends-1.0.25.orig/backend/pieusb_buffer.c sane-backends-1.0.25/backend/pieusb_buffer.c
--- sane-backends-1.0.25.orig/backend/pieusb_buffer.c	2015-09-01 02:31:33.000000000 +0200
+++ sane-backends-1.0.25/backend/pieusb_buffer.c	2015-11-01 15:18:50.000000000 +0100
@@ -100,14 +100,11 @@
 #include <stdio.h>
 #include <fcntl.h>
 #include <sys/mman.h>
-#include <endian.h>
 
-/* When creating the release backend, make complains about unresolved external
- * le16toh, although it finds the include <endian.h> */
 #if __BYTE_ORDER == __LITTLE_ENDIAN
  #define le16toh(x) (x)
 #else
- #define le16toh(x) __bswap_16 (x)
+ #define le16toh(x) ((((x) & 0xff00) >> 8) | (((x) & 0x00ff) << 8))
 #endif
 
 static void buffer_update_read_index(struct Pieusb_Read_Buffer* buffer, int increment);
